### PR TITLE
Nested fillable array

### PR DESCRIPTION
**Pull request by alexanderkraemer**
This PR adds the possibility to make the `$fillable` variable in a mongodb model nested.
So instead of manually validating the input, laravel's notation can be used for this.

Example:
```  
  public $fillable = [
    "a",
    "b",
    "c" => [
      "ca",
      "cb" => [
        "cba"
      ],
      "cc"
    ]
  ];

  public $attributes = [
    "a" => "value a",
    "b" => "value b",
    "d" => "value d",
    "c" => [
      "ca" => "value ca",
      "cs" => "value cs",
      "cb" => [
        "cba" => "value cba",
        "cbb" => "value cbb"
      ],
      "cc" => "value cc"
    ]
  ];
```

The `$attribues` variable mocks the attributes, that are delivered via the `$request` variable in Laravel.
The mass-assignment validation results in the folowing:

```Array
(
    [a] => value a
    [b] => value b
    [c] => Array
        (
            [ca] => value ca
            [cb] => Array
                (
                    [cba] => value cba
                )

            [cc] => value cc
        )

)
```

### CODE REVIEW

**Comment by alexanderkraemer** [comment]
@jenssegers How do you think about that feature?
I currently use it together with the mongodb setup, and it works flawlessly!

**Comment by divine** [comment]
Hello!

Could you please update with the base branch or we can take a look?

Thank you for your contribution!

File: src/Jenssegers/Mongodb/Eloquent/Model.php [hunk 1]
```diff
@@ -43,6 +43,82 @@ abstract class Model extends BaseModel
...
+     * Filter the Attributes given in the request with the $fillable variable from the model,
+     * and only pass on the ones provided in the model.
+     *
+     * @param  array  $filter
+     * @param  array  $attributes
+     * @param  array  &$output
+     *
+     * @throws \Illuminate\Database\Eloquent\MassAssignmentException
+     */
+    private function filterAttributesRecursive($filter, $attributes, &$output)
```
Comment (Smolevich) [inline src/Jenssegers/Mongodb/Eloquent/Model.php]: Using by ref looks like redundant, you can work with `$this->output`

File: src/Jenssegers/Mongodb/Eloquent/Model.php [hunk 2]
```diff
@@ -43,6 +43,82 @@ abstract class Model extends BaseModel
...
+     *  Check wether an element is in the array or not.
+     *  This method flips the key/value pair, if the value is an array.
+     *
+     * @param array $array
+     * @param string $key
+     *
+     */
+    private function isInArray($array, $key)
+    {
+        foreach ($array as $k=>$v) {
```
Comment (Smolevich) [inline src/Jenssegers/Mongodb/Eloquent/Model.php]: Maybe 
```
return in_array($key, array_keys($array)) || in_array($key, array_values($array));
```

**Comment by Smolevich** [comment]
@alexanderkraemer tests are failed, check please

**Comment by divine** [comment]
Closing for now, due to inactivity from the author.

I also don't see the reason of this PR at all.

**Pull request closed without merging**